src/cli/ionic-config.ts

```diff
@@ -0,0 +1,45 @@
+import { readJSON, writeJSON, mkdirp } from '@ionic/utils-fs';
+
+import * as os from 'os';
+import * as path from 'path';
+
+import { uuidv4 } from './telemetry/helpers';
+
+export const CONFIG_FILE = 'config.json';
+export const DEFAULT_CONFIG_DIRECTORY = path.resolve(os.homedir(), '.ionic', CONFIG_FILE);
+
+export interface SystemConfig {
+	"telemetry"?: boolean,
+	"telemetry.stencil"?: boolean,
+	"tokens.telemetry"?: string,
+}
+
+export async function readConfig(): Promise<SystemConfig> {
+	try {
+		return await readJSON(DEFAULT_CONFIG_DIRECTORY);
+	} catch (e) {
+		if (e.code !== 'ENOENT') {
+			throw e;
+		}
+
+		const config: SystemConfig = {
+			"tokens.telemetry": uuidv4(),
+			"telemetry": true,
+			"telemetry.stencil": true,
+		};
+
+		await writeConfig(config);
+
+		return config;
+	}
+}
+
+export async function writeConfig(config: SystemConfig): Promise<void> {
+	await mkdirp(path.dirname(DEFAULT_CONFIG_DIRECTORY));
```

Comment:
I think we  need to ensure two things here:
1. The call to `mkdirp` isn't rejected
2. If the call to `mkdirp` _is_ rejected, that we clean up after ourselves in the event that we clean up after ourselves. From the `mkdirp` [doc](https://github.com/isaacs/node-mkdirp#mkdirpdir-opts---promisestring--undefined)
> Promise rejects if any errors are encountered. Note that, in the case of promise rejection, some directories may have been created, as recursive directory creation is not an atomic operation.

However, the latter can tricky if we intend to reuse this directory for other projects, so perhaps only the first suggestion is required?

Reply:
I'll add recovery for option 1, which would also add messaging for option 2. FWIW, I copied this function directly from ionic-cli, and it's usage is nearly identical.